storage: set shared storage creator ID

RaduBerinde opened the pull request:
Set the Pebble shared object creator ID when initializing or starting
the store. The creator ID is only set if shared storage is configured
from the store (currently only the case with an experimental flag). We
also add a version gate to avoid accidentally setting the creator ID
on a 22.2 store.

Release note: none
Epic: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/97859)
<!-- Reviewable:end -->

jbowens reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 9 of 9 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/97859#-:-NPd5xSV448-u_0GsVKm:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+

RaduBerinde committed:
storage: set shared storage creator ID

Set the Pebble shared object creator ID when initializing or starting
the store. The creator ID is only set if shared storage is configured
from the store (currently only the case with an experimental flag). We
also add a version gate to avoid accidentally setting the creator ID
on a 22.2 store.

Release note: none
Epic: none

RaduBerinde commented:
bors r+
